batch/batch_return.go

```diff
@@ -0,0 +1,48 @@
+package batch
+
+import "github.com/filecoin-project/go-state-types/exitcode"
+
+type BatchReturn struct {
```

Comment:
Maybe add a validate check to the interface? This way we can be sure the data we're working with is good.  Two checks I see are that we're getting monotonic increasing fail codes and all gaps between fail codes are covered by success count.